extensions/2.0/Vendor/EXT_materials_specular_edge_color/README.md

```diff
@@ -0,0 +1,154 @@
+
+The following images show specular color increasing from [0,0,0] to [1,1,1] for `KHR_materials_specular` and for this new extension.
+
+#### KHR_materials_specular
+![](figures/defaultDielectricWhite.png)
+#### EXT_materials_specular_edge_color
+![](figures/edgeColorDielectricWhite.png)
+
+Here is the same comparison for colored specular, increasing from [0,0,0] to [1,0,0] (bottom).
+
+#### KHR_materials_specular
+![](figures/defaultDielectricColor.png)
+#### EXT_materials_specular_edge_color
+![](figures/edgeColorDielectricColor.png)
+
+The specular color factor is allowed to be set to values greater than [1, 1, 1]. Thus, the reflection amount can go beyond what is determined by the index of refraction (IOR). To still ensure energy conservation, the product of specular color factor, specular color texture, and f0 reflectance from IOR is clamped to 1. Please refer to [Implementation](#Implementation) for an example on where to place the clamping operation.
+
+### Conductors
+
+In OpenPBR, metals use the [F82 reflectance model](https://academysoftwarefoundation.github.io/OpenPBR/index.html#model/basesubstrate/metal) (see also the Adobe Standard Material [technical specifications](https://helpx.adobe.com/content/dam/substance-3d/general-knowledge/asm/Adobe%20Standard%20Material%20-%20Technical%20Documentation%20-%20May2023.pdf) for information on F82).  This allows the specular color to be applied at grazing angles to simulate metals with complex IOR's. Many real-world metals exhibit dips in brightness or subtle hue changes at grazing angles.
+
+Here are three real-world metals and one made up, "Unobtainium", to demonstrate the edge-coloring.
+![](figures/edgeColorMetals.png)
+
+For real-world metals, the effect can be hard to see. To make it more visible, here are the results of rendering in a white furnace.
+![](figures/edgeColorMetalsFurnace.png)
+
+This model adds an extra term to the Schlick approximation to accomplish this. 
+
+Let $\mu$ be the cosine of the incident angle and $\overline{\mu}$ is $\cos(82) \approx \frac{1}{7}$. $F(\overline{\mu})$ is the specular color at $F_{{82}}$:
+
+From the OpenPBR specs:
+![](figures/F82-formula.jpg)
+
+## Implementation
+
+*This section is non-normative.*
+
+[Appendix B](https://www.khronos.org/registry/glTF/specs/2.0/glTF-2.0.html#appendix-b-brdf-implementation) defines the function `fresnel_mix`. In this extension, apply this function to the dielectric lobe only and we add two additional arguments called `weight` and `specular_color`.
+
+```
+function fresnel_mix(specular_color, ior, weight, base, layer) {
+  f0 = ((1-ior)/(1+ior))^2
+  f0 = min(f0, float3(1.0))
+  fr = f0 + (1 - f0)*(1 - abs(VdotH))^5
+  return (1 - weight * fr) * base + weight * fr * specular_color * layer
```

Comment:
I think the base weight needs to consider the `specular_color` for reasons of energy preservation. E.g. low specular_color will lead to low specular contribution that is not compensated by the base layer. Also specular color values > 1.0 need to be handled somehow. Haven't fully thought this through but maybe sth. like 

```
specular_weight =  weight * fr * min(specular_color, float3(1.0))
base_weight = 1 -  max_value(specular_weight)
return base_weight * base + specular_weight * layer
```

Reply:
My understanding is that, in OpenPBR, `specular_color` does not effect the base weight.
From the specs:

> The specular_color parameter modulates the Fresnel factor of Fdielectric, but only for the initial reflection of light incident from above, while the light transmitted from above or below (or reflected from below) is assumed to be unaffected. This is technically unphysical if altered from the default white color (as real dielectrics have a Fresnel factor dependent only on the index of refraction), but can be useful in practice to artificially tint the specular highlight without disturbing other aspects of the light transport, i.e. the reflection due to scattering from the internal medium, or the reflection from below, or the transmission from above or below.

Reply:
I think I've got something that works in the Implementation section now.
I definitely wasn't handling the > 1.0 case correctly.